sdk/storage/ustorageclient/build_all/configs/ustorageclient_functions.cmake

```diff
@@ -0,0 +1,189 @@
+#Copyright (c) Microsoft. All rights reserved.
+#Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#Macro to set C and Cpp standards for a given target
+macro(add_standards)
+    set(CMAKE_C_STANDARD                99)
+    set(CMAKE_C_STANDARD_REQUIRED       OFF)
+    set(CMAKE_C_EXTENSIONS              OFF)
+
+    set(CMAKE_CXX_STANDARD              11)
+    set(CMAKE_CXX_STANDARD_REQUIRED     OFF)
+    set(CMAKE_CXX_EXTENSIONS            OFF)
+endmacro()
+
+#Add windows unittest dll
+function(azstorage_c_windows_unittests_add_dll what_is_building folder)
+
+    link_directories(${what_is_building}_dll $ENV{VCInstallDir}UnitTest/lib)
+
+    #Add static library target
+    add_library(${what_is_building}_testsonly_lib STATIC
+            ${${what_is_building}_test_files}
+    )
+
+    #Set the output folder for the target
+    set_target_properties(${what_is_building}_testsonly_lib
+               PROPERTIES
+               FOLDER ${folder} )
+
+    target_include_directories(${what_is_building}_testsonly_lib PUBLIC ${sharedutil_include_directories} $ENV{VCInstallDir}UnitTest/include)
+    target_compile_definitions(${what_is_building}_testsonly_lib PUBLIC CPP_UNITTEST)
+    target_compile_options(${what_is_building}_testsonly_lib PUBLIC /TP /EHsc)
```

Comment:
Hiding /TP within this function is likely to cause a great deal of confusion down the line. Just don't enable the CXX language if you don't want CXX.

Also /EHsc is added by default.